designs/2022-docs-information-architecture-update/README.md

```diff
@@ -0,0 +1,351 @@
+
+- [NO CHANGE] Maintain ESLint section landing page
+  - Current page: [Maintainer Guide](https://eslint.org/docs/latest/maintainer-guide)
+  - Hidden in sidebar
+- [MOVE PAGE] Architecture
+  - Current page: [Architecture](https://eslint.org/docs/latest/developer-guide/architecture/)
+  - Note: in a previous conversation with Nicholas, he mentioned to me that this page is pretty out-of date.
+  - There should probably be a future unit of work to refactor this page.
+  - However, it’s not included in the information architecture updates.
+- [MOVE & REFACTOR PAGE] Set up Development Environment
+  - Consolidate the pages [Getting the Source Code](https://eslint.org/docs/latest/developer-guide/source-code) and [Set up a Development Environment](https://eslint.org/docs/latest/developer-guide/development-environment)
+  - Getting the source code is basically just a step in the process of setting up the development environment. Therefore, these two pages should be consolidated
+  - Also the Directory Structure section on the [Getting the Source Code](https://eslint.org/docs/latest/developer-guide/source-code#directory-structure) page should be moved toward the bottom of the new Set up a Development Environment page.
+- [NEW PAGE(S)] Any other technical pages for maintainers
+  - I don’t have any thoughts on what these should be right now. I would defer to the core maintainers on what they think other maintainers should know.
+  - I think that these additional technical pages would make sense to include in **Phase 5: “Maintain ESLint” documentation updates**of [the documentation update project](https://github.com/eslint/eslint/issues/16365).
+  - NOTE: Maybe should be added, but not part of the initial IA project.
+- [MOVE PAGE] Run the Tests
+  - Current page: [Unit Tests](https://eslint.org/docs/latest/developer-guide/unit-tests)
+- [MOVE & RENAME PAGE] Manage Issues
+  - Current page: [Managing Issues](https://eslint.org/docs/latest/maintainer-guide/issues)
+- [MOVE & RENAME PAGE] Review Pull Requests
+  - Current page: [Reviewing Pull Requests](https://eslint.org/docs/latest/maintainer-guide/pullrequests)
+- [MOVE & RENAME PAGE] Manage Releases
+  - Current page: [Managing Releases](https://eslint.org/docs/latest/maintainer-guide/releases)
+
+### New Sidebar (summary)
+
+With the above changes, the left-hand side navigation will look like:
+
+- Use ESLint in Your Project (landing page)
+  - Getting Started
+  - Core Concepts
+  - Configure
+    - Configuration Files (New)
+    - Configuration Files
+    - Configure Language Options
+    - Configure Rules
+    - Configure Plugins
+    - Ignore Code
+  - Command Line Interface Reference
+  - Rules Reference
+  - Integrations
+  - Migrate to v8.x
+- Extend ESLint (landing page)
+  - Create Plugins
+  - Custom Rules
+  - Custom Formatters
+  - Custom Parsers
+  - Custom Processors
+  - Share Configurations
+  - Node.js API Reference
+- Contribute to ESLint (landing page)
```

Comment:
Does "landing page" mean that "Contribute to ESLint" will be a link in the left-hand side navigation? (pointing to a page with [this content](https://eslint.org/docs/latest/developer-guide/contributing/)).

Reply:
it will be that page with the same content, but given the current way the sidebar works, there will not be a clickable link there. this was a decision made in the past for accessibility purposes. 

nicholas and i were discussing the matter in [this comment thread](https://github.com/eslint/rfcs/pull/97#discussion_r1007459580).

Reply:
Okay, then we should check if there's any important information that only exists on this page because the page (and therefore its content) will effectively be hidden since it won't be in the navigation anymore.

In particular:

 * We should probably have a new page for the [Code of Conduct](https://eslint.org/docs/latest/developer-guide/contributing/#read-the-code-of-conduct) section. Perhaps we could also keep the first paragraph from the current [contributing](https://eslint.org/docs/latest/developer-guide/contributing/) page on this page?
* The CLA section could be merged into the Submit a Pull Request page. There's actually already something [here](https://eslint.org/docs/latest/developer-guide/contributing/pull-requests#step-7-send-the-pull-request), but with fewer details.

This RFC already proposes a new page for [reporting a security vulnerability](https://eslint.org/docs/latest/developer-guide/contributing/#reporting-a-security-vulnerability) so that's covered. Other sections do not seem to provide important additional info.

Reply:
added this content in these 2 PRs:
- 2ca6ee9
- d91afa0